components/formats-bsd/src/loci/formats/tiff/TiffParser.java

```diff
@@ -672,15 +672,11 @@ else if (map instanceof int[]) {
 
     if (ifd.get(IFD.STRIP_BYTE_COUNTS) instanceof OnDemandLongArray) {
       OnDemandLongArray counts = (OnDemandLongArray) ifd.get(IFD.STRIP_BYTE_COUNTS);
-      if (counts != null && counts.getStream() == null) {
```

Comment:
The `counts` field is guaranteed to not be null then? or should the `counts != null` remain?